src/services/WolfDen.Application/Requests/Commands/LeaveManagement/LeaveBalances/InitializeLeaveBalance/InititializeLeaveBalanceCommandHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using MediatR;
+using Microsoft.EntityFrameworkCore;
+using WolfDen.Domain.Entity;
+using WolfDen.Infrastructure.Data;
+
+namespace WolfDen.Application.Requests.Commands.LeaveManagement.LeaveBalances.InitializeLeaveBalance
+{
+    public class InititializeLeaveBalanceCommandHandler(WolfDenContext context) : IRequestHandler<InitializeLeaveBalanceCommand, bool>
+    {
+        private readonly WolfDenContext _context = context;
+
+        public async Task<bool> Handle(InitializeLeaveBalanceCommand request, CancellationToken cancellationToken)
+        {
+            int result;
+            List<LeaveType> leaveType = await _context.LeaveType.ToListAsync(cancellationToken);
+            Employee emp = await _context.Employees.Where(x => x.Id.Equals(request.RequestId)).FirstOrDefaultAsync(cancellationToken);
+            if (emp.JoiningDate.HasValue)
+            {
+                foreach (LeaveType type in leaveType)
+                {
+                    LeaveBalance leaveBalance = new LeaveBalance(request.RequestId, type.Id, 0);
```

Comment:
Purpose of this

Reply:
inorder to initialise leave balance of each leave type when a user is created and updated with their joining date

Reply:
what's point of setting the balance to zero without any business checks

Reply:
Its inorder to get all  the leavetypes updated  for a new employee. Its later on updated using updatebalance command